neotron-bmc-pico/src/main.rs

```diff
@@ -583,7 +589,34 @@ mod app {
 					);
 				}
 			}
-			// TODO: Read ADC for 3.3V and 5.0V rails and check good
+			// TODO: Also monitor 5.0V rail
+
+			// Wait for power-good
+			if ctx.shared.state_dc_power_enabled.lock(|r| *r) == DcPowerState::Starting {
+				let mon_3v3 = ctx.local.adc.read_abs_mv(ctx.local.pin_3v3_monitor);
+				defmt::trace!("3v3 reading: {}", mon_3v3);
+				if mon_3v3 < 1600 {
```

Comment:
Why 1600mV? That seems quite low?

Reply:
There is a voltage divider before the ADC. 1600mV on the pin means 3.2V on the 3.3V rail.
![image](https://github.com/user-attachments/assets/94d82e8d-30b6-4a43-9899-097dbc805898)

So 1600mV may actually be too high. I didn't do an estimate of the tolerances.

Reply:
(If you agree with the general approach, I can work on the details, like adding some explanations, putting the "magic value" 1600 in a `const` etc.)

Reply:
Ahhhh. Yes, you can't exceed fairly the limits on the ADC pin so I halved the input.

Yes the approach makes sense.